From 478314a90e4c8db40c483d7ad5f9337fc561d3ad Mon Sep 17 00:00:00 2001
From: Will Woods <wwoods@redhat.com>
Date: Wed, 14 Mar 2012 17:09:49 -0400
Subject: [PATCH] Make sure 'set -x' gets turned back on in wait_for_loginit

wait_for_loginit does set +x (to turn off debugging temporarily), but
sometimes it would return before turning it back on. Move the set +x
line to fix that, then use 'setdebug' to make sure we don't turn it back
on unless it was needed.
---
 modules.d/99base/dracut-lib.sh |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules.d/99base/dracut-lib.sh b/modules.d/99base/dracut-lib.sh
index f0d426c..6f506b6 100755
--- a/modules.d/99base/dracut-lib.sh
+++ b/modules.d/99base/dracut-lib.sh
@@ -756,9 +756,9 @@ need_shutdown() {
 
 wait_for_loginit()
 {
-    set +x
     [ "$RD_DEBUG" = "yes" ] || return
     [ -e /run/initramfs/loginit.pipe ] || return
+    set +x
     echo "DRACUT_LOG_END"
     exec 0<>/dev/console 1<>/dev/console 2<>/dev/console
         # wait for loginit
@@ -778,7 +778,7 @@ wait_for_loginit()
         kill $(while read line;do echo $line;done</run/initramfs/loginit.pid)
     fi
 
-    set -x
+    setdebug
     rm -f /run/initramfs/loginit.pipe /run/initramfs/loginit.pid
 }
 
